src/program.cpp

```diff
@@ -59,6 +61,7 @@ struct program_impl
     std::unordered_map<std::string, module> modules;
     context ctx;
     std::string target_name;
+    std::vector<context> contexts;
```

Comment:
So from #1751 I also store the targets(and I remove the target_name), but this requires updates to the serialization.

Also this is missing updates to the `assign` method.